### PR TITLE
Verify RABFE Mol Pairs

### PR DESCRIPTION
**Pull request by badisa**
Covers some of the issues run into with RABFE simulations.

Also adds logging for where exactly exceptions came from, which helped find the issues that this attempts to address

### CODE REVIEW

File: fe/model_utils.py [hunk 1]
```diff
@@ -4,7 +4,20 @@
...
+    mol_copy = Chem.AddHs(mol)
+    assert atoms == mol_copy.GetNumAtoms(), "Hydrogens missing for mol"
+
+
+def verify_rabfe_pair(mol: Chem.Mol, blocker: Chem.Mol):
+    assert_mol_has_all_hydrogens(mol)
+    assert_mol_has_all_hydrogens(blocker)
+    mol_charge = Chem.GetFormalCharge(mol)
+    blocker_charge = Chem.GetFormalCharge(blocker)
+    assert mol_charge == blocker_charge, f"Formal charge disagrees: blocker: {blocker_charge:d} ligand: {mol_charge:d}"
```
Comment (maxentile) [inline fe/model_utils.py:20]: Might comment if there are conditions we should make this line more or less stringent.

(Most stringent: `mol_charge == 0, blocker_charge == 0`, less stringent: `mol_charge == blocker_charge`, `abs(mol_charge - blocker_charge) < 2` etc.)

**Review by maxentile — APPROVED** [review]

File: fe/model_utils.py (hunk 1 above)
Comment (badisa) [reply fe/model_utils.py:20]: Based on discussions, I thought we wanted to avoid any 'charge hopping'.

**Review by mcwitt — COMMENTED** [review]
lgtm!

**Review by proteneer — APPROVED** [review]
lgtm! may be good to also check for RMSD later to ensure alignment

**Pull request merged**